idea_town/frontend/static-src/app/main.js

```diff
@@ -18,32 +20,50 @@ app.extend({
     fetch('/api/me?format=json', {
       credentials: 'same-origin'
     }).then((response) => response.json()).then((userData) => {
+      app.on('webChannel:addon-available', () => {
+        app.me.hasAddon = true;
+        app.experiments = new ExperimentsCollection();
+        app.experiments.fetch({
+          success: app.blastOff
+        });
+      });
       app.me = new Me({
         user: userData
       });
+    }).catch((err) => {
+      // for now, log the error in the console & do nothing in the UI
+      console && console.error(err); // eslint-disable-line no-console
+    });
+  },
 
-      app.experiments = new ExperimentsCollection();
-      app.experiments.fetch();
-
-      // session won't change without a hard refresh, but addon state could, so:
-      // if addon state changes, dump user back to '/' and let the router handle
-      // redirecting to the correct landing page
-      app.me.on('change:hasAddon', () => { app.router.reload(); });
+  blastOff() {
+    app.pageManager = new PageManager({
+      pageContainer: document.querySelector('[data-hook=page-container]')
+    });
 
-      app.pageManager = new PageManager({
-        pageContainer: document.querySelector('[data-hook=page-container]')
-      });
+    app.router = new Router();
```

Comment:
Seems like you don't need to create a Router here, [since it's getting done back on line 15 in the `app.extend`](https://github.com/mozilla/idea-town/pull/142/files#diff-9fca6c8d9a32a0b5c86a4223cfac57c3R15)